pkgs/tools/admin/google-cloud-sdk/default.nix

```diff
@@ -3,13 +3,20 @@
 with python27Packages;
 
 stdenv.mkDerivation rec {
-  version = "106.0.0";
   name = "google-cloud-sdk-${version}";
-
-  src = fetchurl {
-    url = "https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-106.0.0-linux-x86_64.tar.gz";
-    sha256 = "00jhpx32sfxcgl404plmb8122bs0ijl2rv25h17mnjn067nhz7nn";
-  };
+  version = "109.0.0";
+
+  src =
+    if stdenv.system == "i686-linux" then
+      fetchurl {
+        url = "https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/${name}-linux-x86.tar.gz";
+        sha256 = "0jcg4pzqmkmpcp86rdi9hcqvvm61rqvl8spl2r1n4658w48h61x7";
+      }
+    else
```

Comment:
I know ! I was being a bit lazy :)

Do you have any good trick to scale the number of uploads to make the sha256 calculation O(1) instead of O(N) ?